src/licenses.ts

```diff
@@ -0,0 +1,35 @@
+import {Change, ChangeSchema} from './schemas'
+
+export function hasInvalidLicenses(
+  changes: Array<Change>,
+  allowLicenses: Array<string> | undefined,
+  failLicenses: Array<string> | undefined
+): Array<Change> {
+  let disallowed: Change[] = []
+
+  if (allowLicenses === undefined) {
+    allowLicenses = []
+  }
+  if (failLicenses === undefined) {
+    failLicenses = []
+  }
+
+  for (const change of changes) {
+    let license = change.license
+    // TODO: be loud about unknown licenses
+    if (license === null) {
+      continue
+    }
+    if (allowLicenses.length > 0) {
+      if (!allowLicenses.includes(license)) {
+        disallowed.push(change)
+      }
+    } else if (failLicenses.length > 0) {
+      if (failLicenses.includes(license)) {
+        disallowed.push(change)
+      }
+    }
```

Comment:
> So, my reading of the logic here is that empty allowLicenses and denyLicenses lists are ignored as filters?

Correct. I'm hesitant to use null because the following is not obvious to me:

> `allowLicenses = []` would mean the obvious: the list of allowed licenses is empty, aka no licenses are allowed.

I was using `null` originally, but I had to insert a few checks to satisfy the compiler and ended up with a version longer than what I have above. 

I don't have strong feelings regarding this, I can change it either here or in a follow-up PR.

Reply:
@febuiles regarding `allowLicences = []`, I think what it boils down to is I don't think `[]` should have a distinctly different meaning to a non-empty array. Consider the following:

```
allow_licenses: []
```

The most natural interpretation of the above, in my view, is "no licenses are allowed". But under the implementation you are proposing, this would behave as "all licenses are allowed". As a user, I have to know that the action is treating `[]` as synonymous with the property not being specified to make sense of the behaviour. I'd rather avoid the potential confusion.